### PR TITLE
Add more funnels for authentication and validate OTP expiration on the server

### PR DESCRIPTION
**Pull request by jieuncodes**
youredith3@gmail.com 으로 이메일 가입 가능!

## Summary by CodeRabbit

- **New Features**
  - Enhanced sign-up process with animated steps and success indicators.
  - Improved form input handling and user feedback in Email, Login, and Sign-Up forms.

- **Bug Fixes**
  - Refined error messages and OTP expiration handling.
  - Corrected API endpoints and error handling in forms.

- **Style**
  - Updated form input styling for better visual presentation.
  - Centered title styling in funnel forms.

- **Performance**
  - Increased rate limit for authentication endpoints.

- **Tests**
  - Added comprehensive tests for OTP verification and email uniqueness checks.

### CODE REVIEW

**Comment by jieuncodes** [comment]
@coderabbitai review

File: server/src/routers/authRoutes.ts [hunk 1]
```diff
@@ -44,19 +44,31 @@ router.post(
   asyncHandler(async (req: AuthRequest, res: Response, next: NextFunction) => {
     const errors = validationResult(req);
     if (!errors.isEmpty()) {
-      return sendErrorResponse(res, 400, "Validation failed");
```
Comment (tedigom52) [inline server/src/routers/authRoutes.ts:47]: sendErrorResponse 함수로 에러처리하던거 확인

File: server/src/routers/authRoutes.ts [hunk 2]
```diff
@@ -69,7 +81,7 @@ router.post(
   }),
   updatePoints,
   (req: AuthRequest, res: Response) => {
-    sendSuccessResponse(res, "User registered");
```
Comment (tedigom52) [inline server/src/routers/authRoutes.ts:72]: sendSuccessResponse로 성공처리하던거 확인

File: server/src/routers/authRoutes.ts [hunk 3]
```diff
@@ -190,8 +195,12 @@ router.post(
       return sendErrorResponse(res, 400, "Email is required");
     }
 
+    await otpRepository.delete({ email });
+
     const otp = crypto.randomInt(100000, 999999).toString();
     const expirationTime = new Date(Date.now() + 5 * 60 * 1000);
+    console.log("expirationTime", expirationTime);
+
```
[the remark this answers is not in the document]
Comment (jieuncodes) [reply server/src/routers/authRoutes.ts:203]: OMG

File: server/src/routers/authRoutes.ts (hunk 1 above)
Comment (jieuncodes) [reply server/src/routers/authRoutes.ts:47]: @tedigom52 확인 부탁!

**Pull request merged**